### ISSUE TITLE
Cannot group by tags in Netbox v2.9.3

### ISSUE THREAD

**Issue by inigoleaga** [netbox-community/ansible_modules#346]
##### ISSUE TYPE

 - Bug Report

###### Ansible:
2.9.11

###### Netbox:
2.9.3

###### Collection:
Don't know... How can I check it?

##### SUMMARY
A yaml file is used as an inventory source in AWX. Specifying "tags" in the "group by" field gives a synchronization error. Apparently it expects a string to be returned and yet it receives a dictionary:

##### STEPS TO REPRODUCE
In AWX try to update the following "netbox_inventory.yml" as an Inventory Source:

```
# Example command line: ansible-inventory -v --list -i netbox_inventory.yml
plugin: netbox.netbox.nb_inventory
api_endpoint: https://XXXXXXXXXXXX
validate_certs: False
token: XXXXXXXXXXXXXXXXXXXXXXX
#config_context: True
#interfaces: True
plurals: True
group_names_raw: True
group_by:
  - device_roles
  - sites
  - manufacturers
  - tags
query_filters:
  - tag: awx
#compose:
#  id: id
```

##### EXPECTED RESULTS
A good synchronization log and the devices grouped by the assigned tag in netbox as in previous versions

##### ACTUAL RESULTS
Synchronization error... Pay attention to the first warning.

```
    2.141 INFO     Updating inventory 8: netbox_inventory
    2.590 DEBUG    Using base command: python /usr/bin/ansible-inventory -i /tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml --playbook-dir /tmp/awx_137838_bu7awdwa/project -vvvvv
    2.590 INFO     Reading Ansible inventory source: /tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml
    2.592 INFO     Using VIRTUAL_ENV: /var/lib/awx/venv/ansible
    2.592 INFO     Using PATH: /var/lib/awx/venv/ansible/bin:/var/lib/awx/venv/awx/bin:/usr/pgsql-10/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    2.592 INFO     Using PYTHONPATH: /var/lib/awx/venv/ansible/lib/python3.6/site-packages:
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/bin/awx-manage", line 8, in <module>
    sys.exit(manage())
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/awx/__init__.py", line 154, in manage
    execute_from_command_line(sys.argv)
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1149, in handle
    raise exc
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1039, in handle
    venv_path=venv_path, verbosity=self.verbosity).load()
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 215, in load
    return self.command_to_json(base_args + ['--list'])
  File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 198, in command_to_json
    self.method, proc.returncode, stdout, stderr))
RuntimeError: ansible-inventory failed (rc=1) with stdout:
stderr:
ansible-inventory 2.9.11
  config file = /tmp/awx_137838_bu7awdwa/project/ansible.cfg
  configured module search path = ['/var/lib/awx/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible-inventory
  python version = 3.6.8 (default, Apr 16 2020, 01:36:27) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]
Using /tmp/awx_137838_bu7awdwa/project/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml as it did not pass its verify_file() method
script declined parsing /tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml as it did not pass its verify_file() method
Fetching: https://struc.sener.es/api/docs/?format=openapi
Fetching: https://struc.sener.es/api/dcim/devices/?limit=0&tag=awx&exclude=config_context
Fetching: https://struc.sener.es/api/virtualization/virtual-machines/?limit=0&tag=awx&exclude=config_context
Fetching: https://struc.sener.es/api/dcim/sites/?limit=0
Fetching: https://struc.sener.es/api/dcim/regions/?limit=0
Fetching: https://struc.sener.es/api/tenancy/tenants/?limit=0
Fetching: https://struc.sener.es/api/dcim/racks/?limit=0
Fetching: https://struc.sener.es/api/dcim/rack-groups/?limit=0
Fetching: https://struc.sener.es/api/dcim/device-roles/?limit=0
Fetching: https://struc.sener.es/api/dcim/platforms/?limit=0
Fetching: https://struc.sener.es/api/dcim/device-types/?limit=0
Fetching: https://struc.sener.es/api/dcim/manufacturers/?limit=0
Fetching: https://struc.sener.es/api/virtualization/clusters/?limit=0
Fetching: https://struc.sener.es/api/ipam/services/?limit=0
toml declined parsing /tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml as it did not pass its verify_file() method
Fetching: https://struc.sener.es/api/docs/?format=openapi
Fetching: https://struc.sener.es/api/dcim/devices/?limit=0&tag=awx&exclude=config_context
Fetching: https://struc.sener.es/api/virtualization/virtual-machines/?limit=0&tag=awx&exclude=config_context
Fetching: https://struc.sener.es/api/dcim/sites/?limit=0
Fetching: https://struc.sener.es/api/dcim/regions/?limit=0
Fetching: https://struc.sener.es/api/tenancy/tenants/?limit=0
Fetching: https://struc.sener.es/api/dcim/racks/?limit=0
Fetching: https://struc.sener.es/api/dcim/rack-groups/?limit=0
Fetching: https://struc.sener.es/api/dcim/device-roles/?limit=0
Fetching: https://struc.sener.es/api/dcim/platforms/?limit=0
Fetching: https://struc.sener.es/api/dcim/device-types/?limit=0
Fetching: https://struc.sener.es/api/dcim/manufacturers/?limit=0
Fetching: https://struc.sener.es/api/virtualization/clusters/?limit=0
Fetching: https://struc.sener.es/api/ipam/services/?limit=0
[WARNING]:  * Failed to parse
/tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml with auto plugin: Invalid
group name supplied, expected a string but got <class 'dict'> for {'id': 129,
'url': 'https://struc.sener.es/api/extras/tags/129/', 'name': 'access', 'slug':
'access', 'color': '00bcd4'}
  File "/usr/lib/python3.6/site-packages/ansible/inventory/manager.py", line 280, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/usr/lib/python3.6/site-packages/ansible/plugins/inventory/auto.py", line 58, in parse
    plugin.parse(inventory, loader, path, cache=cache)
  File "/tmp/awx_137838_bu7awdwa/requirements_collections/ansible_collections/netbox/netbox/plugins/inventory/nb_inventory.py", line 1383, in parse
    self.main()
  File "/tmp/awx_137838_bu7awdwa/requirements_collections/ansible_collections/netbox/netbox/plugins/inventory/nb_inventory.py", line 1342, in main
    self.add_host_to_groups(host=host, hostname=hostname)
  File "/tmp/awx_137838_bu7awdwa/requirements_collections/ansible_collections/netbox/netbox/plugins/inventory/nb_inventory.py", line 1191, in add_host_to_groups
    transformed_group_name = self.inventory.add_group(group=group_name)
  File "/usr/lib/python3.6/site-packages/ansible/inventory/data.py", line 163, in add_group
    raise AnsibleError("Invalid group name supplied, expected a string but got %s for %s" % (type(group), group))
[WARNING]:  * Failed to parse
/tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml with yaml plugin: Plugin
configuration YAML file, not YAML inventory
  File "/usr/lib/python3.6/site-packages/ansible/inventory/manager.py", line 280, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/usr/lib/python3.6/site-packages/ansible/plugins/inventory/yaml.py", line 112, in parse
    raise AnsibleParserError('Plugin configuration YAML file, not YAML inventory')
[WARNING]:  * Failed to parse
/tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml with ini plugin: Invalid
host pattern 'plugin:' supplied, ending in ':' is not allowed, this character
is reserved to provide a port.
  File "/usr/lib/python3.6/site-packages/ansible/inventory/manager.py", line 280, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/usr/lib/python3.6/site-packages/ansible/plugins/inventory/ini.py", line 138, in parse
    raise AnsibleParserError(e)
[WARNING]:  * Failed to parse
/tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml with
ansible_collections.netbox.netbox.plugins.inventory.nb_inventory plugin:
Invalid group name supplied, expected a string but got <class 'dict'> for
{'id': 129, 'url': 'https://struc.sener.es/api/extras/tags/129/', 'name':
'access', 'slug': 'access', 'color': '00bcd4'}
  File "/usr/lib/python3.6/site-packages/ansible/inventory/manager.py", line 280, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/tmp/awx_137838_bu7awdwa/requirements_collections/ansible_collections/netbox/netbox/plugins/inventory/nb_inventory.py", line 1383, in parse
    self.main()
  File "/tmp/awx_137838_bu7awdwa/requirements_collections/ansible_collections/netbox/netbox/plugins/inventory/nb_inventory.py", line 1342, in main
    self.add_host_to_groups(host=host, hostname=hostname)
  File "/tmp/awx_137838_bu7awdwa/requirements_collections/ansible_collections/netbox/netbox/plugins/inventory/nb_inventory.py", line 1191, in add_host_to_groups
    transformed_group_name = self.inventory.add_group(group=group_name)
  File "/usr/lib/python3.6/site-packages/ansible/inventory/data.py", line 163, in add_group
    raise AnsibleError("Invalid group name supplied, expected a string but got %s for %s" % (type(group), group))
[WARNING]: Unable to parse
/tmp/awx_137838_bu7awdwa/project/netbox_inventory.yml as an inventory source
ERROR! No inventory was parsed, please check your configuration and options.``
```

**Comment by szelga** [comment]
should be addressed in #340

**Comment by inigoleaga** [comment]
Thanks @szelga , this is a dup of https://github.com/netbox-community/ansible_modules/issues/287
